stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 Camptocamp SA
+# Copyright 2016 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from collections import defaultdict
+from odoo import api, fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = 'stock.warehouse.orderpoint'
+
+    product_location_qty_available_not_res = fields.Float(
+        string='Quantity On Location (Unreserved)',
+        compute='_compute_product_available_qty',
+    )
+
+    @api.multi
+    def _compute_product_available_qty(self):
+        super()._compute_product_available_qty()
+        op_by_loc = defaultdict(lambda: self.env['stock.warehouse.orderpoint'])
+        for order in self:
+            op_by_loc[order.location_id] |= order
```

Comment:
This pattern of extending the recordset is very slow compared to using a true `set`, then create a new recordset out of the final set. See https://github.com/OCA/connector-prestashop/pull/53/commits/5d617fdbc85b1fdb23a4557b8d456b1b2e43720a